### DISCUSSION TITLE
Best approach for checking out the v2.0 beta multi vendor support

### DISCUSSION THREAD

**Discussion by tlodge** [vendurehq/vendure#2156] [Q&A] [answered]
First of all - a huge thank you for your amazing work - I’m so delighted that 2.0 will have multivendor support.  My question is how best to have a play with it now.  I have installed the beta version of vendure with:

```
npx @vendure/create@next my-shop
```

And that all works fine, and I see that I have a new ‘seller’ in the admin interface.  However when I then try and use the multivendor plugin (which I copy to src/plugins from the vendure repo, with the  commit 9ba67a0d08) and add the MultivendorPlugin to my dev-config.ts, then do a yarn build, I get a typescript compile error, which I assume is a mismatch between the v2.0 beta and the plugin:

```
src/plugins/multivendor-plugin/multivendor.plugin.ts:188:17 - error TS2345: Argument of type '{ code: string; name: string; enabled: true; handler: { code: string; arguments: never[]; }; }' is not assignable to parameter of type 'CreatePaymentMethodInput'.
  Object literal may only specify known properties, and 'name' does not exist in type 'CreatePaymentMethodInput'.

188                 name: 'Connected Payments',
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/plugins/multivendor-plugin/service/mv.service.ts:108:42 - error TS2339: Property 'currencyCode' does not exist on type 'Channel'.

108             currencyCode: defaultChannel.currencyCode,
                                             ~~~~~~~~~~~~


Found 2 errors in 2 files.
```

I then thought that maybe I could try the major branch of the main repo, but when I try and compile that I get a:  

```
✖  @vendure/admin-ui-plugin:build
$ rimraf lib && node -r ts-node/register build.ts && yarn compile
       Building admin-ui from source...
$ yarn ng build vendure-admin --configuration production
$ ng build vendure-admin --configuration production
       The esbuild browser application builder ('browser-esbuild') is currently experimental.
       The 'budgets' option is currently unsupported by this experimental builder and will be ignored.
       The 'progress' option is currently unsupported by this experimental builder and will be ignored.
       ✘ [ERROR] Could not resolve "@angular/router"
       
           ../../node_modules/@clr/angular/fesm2020/clr-angular.mjs:14:22:
             14 │ import * as i1$1 from '@angular/router';
                ╵                       ~~~~~~~~~~~~~~~~~
       
         You can mark the path "@angular/router" as external to exclude it from the bundle, which will remove this error.
```

So I guess my question is whether I should stop being impatient and wait for the v2 release, or whether there is a recommended way to start playing with the multivendor support in the beta release.

Once again, many many thanks for all you have done with this project!

**Comment by michaelbromley** [comment, marked as answer]
Hi,

Thanks for pointing this out - the multivendor plugin hasn't been updated with a couple of breaking API changes from the latest beta releases. From those errors I see the following changes are needed:

1. This one has been fixed in the latest major branch it seems. Update your multivendor.plugin.ts file with these lines: https://github.com/vendure-ecommerce/vendure/blob/d06ac0b8c9ab264f0818af01e36b4cc1aaa57eb0/packages/dev-server/test-plugins/multivendor-plugin/multivendor.plugin.ts#L187C1-L198
2. Change [this line](https://github.com/vendure-ecommerce/vendure/blob/19aa3a04245f709fc58078b4ca92649dd3b7f3c5/packages/dev-server/test-plugins/multivendor-plugin/service/mv.service.ts#L108) to read:
   ```ts
   currencyCode: defaultChannel.defaultCurrencyCode,
   ```

Hopefully that should then compile for you. You'll still find things are missing - especially the admin ui parts are not yet fleshed out. I'm working on that right now :)

**Reply by tlodge** [reply]
Thanks for this! Will do.
